runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/CredentialsProviderChain.kt

```diff
@@ -16,5 +16,8 @@ import aws.smithy.kotlin.runtime.util.Attributes
  */
 public class CredentialsProviderChain(vararg providers: CredentialsProvider) :
     IdentityProviderChain<CredentialsProvider, Credentials>(*providers), CredentialsProvider {
+
+    public constructor(providers: List<CredentialsProvider>) : this(*providers.toTypedArray())
```

Comment:
Missing overload for other provider chains, e.g. [BearerTokenProviderChain](https://github.com/awslabs/smithy-kotlin/blob/53b1a21419b177871607c982ed16c1fa248e084f/runtime/auth/http-auth/common/src/aws/smithy/kotlin/runtime/http/auth/BearerTokenProviderChain.kt#L18)